OpenRA.Game/FileSystem/InstallShieldCABExtractor.cs

```diff
@@ -0,0 +1,440 @@
+			public CommonHeader(BinaryReader reader)
+			{
+				Version = reader.ReadUInt32();
+				VolumeInfo = reader.ReadUInt32();
+				CabDescriptorOffset = (long)reader.ReadUInt32();
+				CabDescriptorSize = reader.ReadUInt32();
+			}
+		}
+
+		struct CabDescriptor
+		{
+			public readonly long FileTableOffset;
+			public readonly uint FileTableSize;
+			public readonly uint FileTableSize2;
+			public readonly uint DirectoryCount;
+			public readonly uint FileCount;
+			public readonly long FileTableOffset2;
+			public CabDescriptor(BinaryReader reader, CommonHeader commonHeader)
+			{
+				reader.BaseStream.Seek(commonHeader.CabDescriptorOffset + 0xC, SeekOrigin.Begin);
+				FileTableOffset = (long)reader.ReadUInt32();
+				reader.ReadUInt32();
+				FileTableSize = reader.ReadUInt32();
+				FileTableSize2 = reader.ReadUInt32();
+				DirectoryCount = reader.ReadUInt32();
+				reader.ReadUInt64();
+				FileCount = reader.ReadUInt32();
+				FileTableOffset2 = (long)reader.ReadUInt32();
+			}
+		}
+
+		struct FileDescriptor
+		{
+			public readonly ushort	Flags;
+			public readonly uint	ExpandedSize;
+			public readonly uint	CompressedSize;
+			public readonly uint	DataOffset;
+			public readonly byte[]	MD5;
+			public readonly uint	NameOffset;
+			public readonly ushort	DirectoryIndex;
+			public readonly uint	LinkPrevious;
+			public readonly uint	LinkNext;
+			public readonly byte	LinkFlags;
+			public readonly ushort	Volume;
+			public readonly string	Filename;
+
+			public FileDescriptor(BinaryReader reader, long tableOffset)
+			{
+				Flags			= reader.ReadUInt16();
+				ExpandedSize		= reader.ReadUInt32();
+				reader.ReadUInt32();
+				CompressedSize		= reader.ReadUInt32();
+				reader.ReadUInt32();
+				DataOffset		= reader.ReadUInt32();
+				reader.ReadUInt32();
+				MD5			= reader.ReadBytes(0x10);
+				reader.ReadBytes(0x10);
+				NameOffset		= reader.ReadUInt32();
+				DirectoryIndex		= reader.ReadUInt16();
+				reader.ReadBytes(0xc);
+				LinkPrevious		= reader.ReadUInt32();
+				LinkNext		= reader.ReadUInt32();
+				LinkFlags		= reader.ReadByte();
+				Volume			= reader.ReadUInt16();
+				var posSave		= reader.BaseStream.Position;
+
+				reader.BaseStream.Seek(tableOffset + NameOffset, SeekOrigin.Begin);
+				Filename = reader.BaseStream.ReadASCIIZ();
+				reader.BaseStream.Seek(posSave, SeekOrigin.Begin);
+			}
+		}
+
+		readonly Stream hdrFile;
+		CommonHeader commonHeader;
+		CabDescriptor cabDescriptor;
+		List<uint> directoryTable;
+		Dictionary<uint, string> directoryNames;
+		Dictionary<uint, FileDescriptor> fileDescriptors;
+		Dictionary<string, FileDescriptor> fileDict;
+		List<uint> fileGroupOffsets;
+		List<FileGroup> fileGroups;
+		int priority;
+		string commonName;
+		public int Priority
+		{
+			get
+			{
+				return priority;
+			}
+		}
+
+		public string Name
+		{
+			get
+			{
+				return commonName;
+			}
+		}
+
+		public InstallShieldCABExtractor(string hdrFilename, int priority_ = -1)
+		{
+			priority = priority_;
```

Comment:
OpenRA conventions don't include '_' anywhere. Rename `priority_` to `priority` and change this line to
`this.priority = priority;`.